### PR TITLE
[HOTFIX] Remove `-g` from cython compile commands

### PR DESCRIPTION
**Pull request by ajschmidt8**
Removes `-g` from the compile commands generated by distutils to compile Cython files.

This will make our container images, conda packages, and python wheels smaller.

### CODE REVIEW

File: python/setup.py [hunk 1]
```diff
@@ -105,42 +140,50 @@ def run(self):
         os.system('find . -name "*.cpython*.so" -type f -delete')
 
 
+class build_ext_no_debug(_build_ext):
```
Comment (rlratzel) [inline python/setup.py:143]: minor: I've always seen "release" used as the term for "no debug" (ie. it's either a release build or a debug build).

**Review by rlratzel — CHANGES_REQUESTED** [review]
Looks good, thanks for doing this. I had one minor naming suggestion, and this will also need to be applied to the latest cugraph changes which add `pylibcugraph` (it has its own setup.py) and change the directory structure. I can help with that if you want.

File: python/setup.py (hunk 1 above)
Comment (rlratzel) [reply python/setup.py:143]: I just noticed this is a hotfix for 21.08, so I don't mind if this change isn't made (I'm fine either way).  I'll approve.

**Review by rlratzel — APPROVED** [review]

File: python/setup.py (hunk 1 above)
Comment (BradReesWork) [reply python/setup.py:143]: @ajschmidt8 is 21.08 and main the correct branches?

**Comment by rlratzel** [comment]
rerun tests

File: python/setup.py (hunk 1 above)
Comment (ajschmidt8) [reply python/setup.py:143]: yes, these are the correct branches. this needs to go into `branch-21.08`.

**Pull request merged**